File: client/src/test/java/com/jraska/github/client/logging/FirebaseCrashReporterTest.java
package com.jraska.github.client.logging;

import org.junit.Test;

import static org.mockito.ArgumentMatchers.eq;
import static org.mockito.Mockito.*;

public class FirebaseCrashReporterTest {
  @Test
  public void whenNoMessageArgument_thenPasses() {
    FirebaseCrashProxy crashProxyMock = mock(FirebaseCrashProxy.class);
    FirebaseCrashReporter crashReporter = new FirebaseCrashReporter(crashProxyMock);
    RuntimeException exception = new RuntimeException();

    crashReporter.report(exception);

    verify(crashProxyMock, only()).report(eq(exception));
  }
}

File: client/src/test/java/com/jraska/github/client/logging/FirebaseCrashReporterTest.kt
package com.jraska.github.client.logging

import org.junit.Test

import org.mockito.ArgumentMatchers.eq
import org.mockito.Mockito.*

class FirebaseCrashReporterTest {
  @Test
  fun whenNoMessageArgument_thenPasses() {
    val crashProxyMock = mock(FirebaseCrashProxy::class.java)
    val crashReporter = FirebaseCrashReporter(crashProxyMock)
    val exception = RuntimeException()

    crashReporter.report(exception)

    verify(crashProxyMock, only()).report(eq(exception))
  }
}
